mlflow/store/artifact/unity_catalog_oss_models_artifact_repo.py

```diff
@@ -0,0 +1,165 @@
+from mlflow.utils.rest_utils import (
+    _REST_API_PATH_PREFIX,
+    _UC_OSS_REST_API_PATH_PREFIX,
+    call_endpoint,
+    extract_api_info_for_service,
+)
+from mlflow.utils.uri import (
+    _OSS_UNITY_CATALOG_SCHEME,
+    get_databricks_profile_uri_from_artifact_uri,
+    get_db_info_from_uri,
+    is_oss_unity_catalog_uri,
+)
+
+_METHOD_TO_INFO = extract_api_info_for_service(UcModelRegistryService, _REST_API_PATH_PREFIX)
+_METHOD_TO_INFO_OSS = extract_api_info_for_service(
+    UnityCatalogService, _UC_OSS_REST_API_PATH_PREFIX
+)
+
+from mlflow.utils.uri import is_file_uri
+from mlflow.store.artifact.local_artifact_repo import LocalArtifactRepository
+
+class UnityCatalogOSSModelsArtifactRepository(ArtifactRepository):
+    """
+    Performs storage operations on artifacts controlled by a Unity Catalog model registry
+
+    Temporary scoped tokens for the appropriate cloud storage locations are fetched from the
+    remote backend and used to download model artifacts.
+
+    The artifact_uri is expected to be of the form `models:/<model_name>/<model_version>`
+
+    Note : This artifact repository is meant is to be instantiated by the ModelsArtifactRepository
+    when the client is pointing to a Unity Catalog model registry.
+    """
+
+    def __init__(self, artifact_uri, registry_uri):
+        if not is_oss_unity_catalog_uri(registry_uri):
+            raise MlflowException(
+                message="Attempted to instantiate an artifact repo to access models in the "
+                f"OSSS Unity Catalog with non-Unity Catalog registry URI '{registry_uri}'. "
+                f"Please specify a Unity Catalog registry URI of the "
+                f"form '{_OSS_UNITY_CATALOG_SCHEME}[://profile]', e.g. by calling "
+                f"mlflow.set_registry_uri('{_OSS_UNITY_CATALOG_SCHEME}') if using the "
+                f"MLflow Python client",
+                error_code=INVALID_PARAMETER_VALUE,
+            )
+        super().__init__(artifact_uri)
+        from mlflow.tracking.client import MlflowClient
+
+        registry_uri_from_artifact_uri = get_databricks_profile_uri_from_artifact_uri(
+            artifact_uri, result_scheme=_OSS_UNITY_CATALOG_SCHEME
+        )
+        if registry_uri_from_artifact_uri is not None:
+            registry_uri = registry_uri_from_artifact_uri
+        _, key_prefix = get_db_info_from_uri(registry_uri)
+        if key_prefix is not None:
+            raise MlflowException(
+                "Remote model registry access via model URIs of the form "
+                "'models://<scope>@<prefix>/<model_name>/<version_or_stage>' is unsupported for "
+                "models in the Unity Catalog. We recommend that you access the Unity Catalog "
+                "from the current Databricks workspace instead."
+            )
+        self.registry_uri = registry_uri
+        self.client = MlflowClient(registry_uri=self.registry_uri)
+        try:
+            spark = _get_active_spark_session()
```

Comment:
Is UC OSS model registry intended to be accessed outside of Spark? 
Is there another utility that can be used to access the model name directly via an API request? 
This will raise a very confusing error if spark isn't installed in a user's environment:
https://github.com/mlflow/mlflow/blob/05485c4c8e615da34eb9fff83a43002ede277226/mlflow/utils/_unity_catalog_utils.py#L262-L281

Reply:
I think the idea is this could be accessed in both spark and non-spark situations?  Why would this throw a user facing error if the exception is caught and passed?  With spark being None, line 272-273 would trigger in the _unity_catalog_utils.py file and I don't think any exception would occur, or am I reading this incorrectly?

Reply:
ahhhh dang I totally missed that conditional on 272 :) 
Tested locally in an environment that doesn't have Spark installed and I have no problem with this :)